draft-jennings-perc-double-01.md

```diff
@@ -96,127 +101,176 @@ Terms:
-in bit field and perhaps packing some of the single values into the same byte.
+If an MDD modifies an original RTP header value, the MDD MUST include the
+OHB extension to reflect the changed value(s).  If another MDD along the
+media path makes additional changes to the RTP header and the original
+value is not already present in the OHB, the MDD must extend the OHB by
+adding the changed value to the OHB.  So as to properly preserve original
+RTP header values, an MDD MUST NOT change a value already present in the
+OHB extension.
 
-# Operations
+# RTP Operations
 
 
 ## Encrypting a Packet
 
 To encrypt a packet, the endpoint encrypts the packet with
-the inner transform, may add an OHB, then applies the outer transform.
+the inner transform and then applies the outer transform.  The processes
+is as follows:
 
 * Form an RTP packet.  If there are any header extensions, they MUST use
   {{RFC5285}}.
 
-* Apply the transform to the RTP packet
+* Apply the inner cryptographic transform to the RTP packet
 
-* Optionally add an OHB header extension.  The endpoint MAY include any header
-  fields that are signaled to be modified by the MDD, to reduce processing
-  burden on the MDD. Open Issue: do we want the sending client to be able to add
-  an OHB?
+* Optionally add an OHB header extension.  The endpoint MAY include the OHB
+  containing copies of RTP header information it knows through some means
+  defined outside of this document that the MDD is likely to change.
+{Note: I don't like this step, because it doesn't really save
+any work for the MDD.  The MDD has to look for the field and get prepared
+to insert it or change it.  By the time it does that, there is really
+very little work left.  And I don't like it being mandatory, either, because
+the only logical mandatory requirement would be the 3-octet variety, thus
+making every packet 3 octets longer.  I prefer we just leave this to the MDD.}
 
-* Apply the SRTP cryptographic transform with the outer parameters (outer
-  transform)
+* Apply the outer cryptographic transform to the RTP packet
 
 
 ## Modifying a Packet
 
 In order to modify a packet, the MDD undoes the outer transform, modifies the
-packet, updates the OHB with any new modifications, and re-applies the outer
-tranform.
-
-* Apply the (outer) decryption transform to the packet
+packet, updates the OHB with any modifications not already present in the OHB,
+and re-applies the outer transform.
 
-* Separate the OHB from the (encrypted) original payload
+* Apply the outer decryption transform to the packet
 
 * Change any required parameters
 
-* If a changed parameter is not already in the OHB, add it with its original
-  value to the OHB. Note that in the case of cascaded MDDs, the first MDD may
-  have already added an OHB.
+* If a changed RTP header field is not already in the OHB, add it with its
+  original value to the OHB.  Note that in the case of cascaded MDDs, the
```

Comment:
Might need to point out this will change the type OHB from say a 2 byte to 3 byte.